lib/utils/suppressionsService.mjs

```diff
@@ -0,0 +1,247 @@
+/**
+ * Inspired by: https://github.com/eslint/eslint/blob/v9.26.0/lib/services/suppressions-service.js
```

Comment:
[suggestion] This file seems not only "Inspired by" but almost a copy of the original file. If yes, we may need to declare the copyright or the author in this file. 🤔 

Ref https://github.com/eslint/eslint/blob/eaf8a418af32b3190494e4a2284533353c28ccfa/LICENSE